cmake/modules/BuildBoost.cmake

```diff
@@ -130,7 +138,19 @@ function(do_build_boost version)
   # edit the user-config.jam so, so b2 will be able to use the specified
   # toolset
   file(WRITE ${source_dir}/user-config.jam
-    "using ${toolset} : : ${CMAKE_CXX_COMPILER} ;")
+    "using ${toolset}"
+    " : "
+    " : ${CMAKE_CXX_COMPILER}"
+    " ;\n")
+  if(with_python GREATER -1)
+    file(APPEND ${source_dir}/user-config.jam
+      "using python"
+      " : ${PYTHON_VERSION_STRING}"
```

Comment:
From http://www.boost.org/doc/libs/1_58_0/libs/python/doc/building.html#python-configuration-parameters

version
    the version of Python to use. Should be in Major.Minor format, for example, 2.3. Do not include the subminor version (i.e. not 2.5.1).

Except `${PYTHON_VERSION_STRING}` expands to a value that DOES include the subminor version:

`PYTHON_VERSION_STRING 3.6.4`

Reply:
Now testing:

```
-      " : ${PYTHON_VERSION_STRING}"
+      " : ${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}"
```